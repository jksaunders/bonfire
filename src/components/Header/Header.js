import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import {
  HeaderButton,
  HeaderItem,
  HeaderLogo
} from "./subcomponents";

const CONSTANTS = {};

const padding = "16px";

const StyledHeader = styled.div`
  align-items: center;
  display: flex;
  height: ${({ height }) => height};
  justify-content: flex-end;
  padding: 0px ${padding};
  
  & > img {
    height: ${({ height }) => height};
    margin-right: auto;
  }
`;

const propTypes = {
  children: PropTypes.node,
  height: PropTypes.string
};

const defaultProps = {
  children: null,
  height: "125px"
};

const Header = ({
  children,
  ...rest
}) => (
  <StyledHeader {...rest}>
    {children}
  </StyledHeader>
);

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;
Header.CONSTANTS = CONSTANTS;
Header.HeaderButton = HeaderButton;
Header.HeaderItem = HeaderItem;
Header.HeaderLogo = HeaderLogo;

export default Header;