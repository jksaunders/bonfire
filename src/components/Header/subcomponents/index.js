export { default as HeaderButton } from "./HeaderButton";
export { default as HeaderItem } from "./HeaderItem";
export { default as HeaderLogo } from "./HeaderLogo";