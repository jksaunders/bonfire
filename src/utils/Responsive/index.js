export { default as ResponsiveRoot } from './ResponsiveRoot';
export * from './ResponsiveRoot';
