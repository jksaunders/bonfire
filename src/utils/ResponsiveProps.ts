import { SizeProp } from './Sizes';

export type ResponsiveProp<T> = T | SizeProp<T>;
export type ResponsiveProps<T, ResponsiveKeys = keyof T> = {
  [K in keyof T]: K extends ResponsiveKeys
    ? T[K] extends undefined
      ? ResponsiveProp<T[K]>
      : ResponsiveProp<NonNullable<T[K]>>
    : T[K];
};
