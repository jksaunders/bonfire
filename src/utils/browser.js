const goTo = (url) => {
  if (url && typeof window !== 'undefined') {
    window.location.href = url;
  }
};

export default {
  goTo
};