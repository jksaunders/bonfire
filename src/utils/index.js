export { default as Browser } from "./browser";
export * from "./styling";