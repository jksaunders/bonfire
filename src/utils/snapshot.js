/* eslint-disable import/no-extraneous-dependencies */
import { render as importedRender } from '@testing-library/react';
import 'jest-styled-components';

export const render = component => {
  const rendered = importedRender(component);
  return {
    ...rendered,
    component: rendered.container.firstChild,
  };
};

export const expectSnapshot = component => {
  expect(component).toMatchSnapshot();
};

export const expectExists = component => {
  expect(component).not.toEqual(null);
  expect(component).toBeInTheDocument();
};

export const expectStyle = (component, key, value) => {
  expect(component).toHaveStyleRule(key, value);
};
