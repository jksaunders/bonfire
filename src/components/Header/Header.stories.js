import React from "react";
import styled, { ThemeProvider } from "styled-components";

import { storiesOf } from "@storybook/react";
import { select, withKnobs } from "@storybook/addon-knobs";

import { ThemeConstants } from "../../theming";
import Header from "./Header";

const StoryHeader = styled(Header)`
  background: linear-gradient(270deg, #48C9B0 0%, white 100%);
`;

storiesOf("Header", module)
  .addDecorator(withKnobs)
  .add("with text", () => {
    const themeMode = select("Theme Mode", Object.keys(ThemeConstants.mode.values), Object.keys(ThemeConstants.mode.values)[0]);
    const themeLayout = select("Theme Layout", Object.keys(ThemeConstants.layout.values), Object.keys(ThemeConstants.layout.values)[0]);
    const theme = { mode: themeMode, layout: themeLayout };

    return (
      <ThemeProvider theme={theme}>
        <div>
          <div>
            <StoryHeader>
              <Header.HeaderItem text="Programs" link="https://gooogle.com" />
              <Header.HeaderItem text="About Us" />
              <Header.HeaderItem text="Events" />
              <Header.HeaderItem text="Contact Us" />
              <Header.HeaderButton text="Donate" variant={Header.HeaderButton.CONSTANTS.VARIANT.PRIMARY} />
            </StoryHeader>
          </div>
        </div>
      </ThemeProvider>
    );
  });