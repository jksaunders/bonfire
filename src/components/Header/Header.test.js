import React from 'react';
import { expectExists, expectSnapshot, render } from '../../utils/snapshot';
import Header from './Header';

const getHeader = (props = {}) => render(<Header {...props} />);

test('Header renders', () => {
  const { component } = getHeader();
  expectExists(component);
  expectSnapshot(component);
});
