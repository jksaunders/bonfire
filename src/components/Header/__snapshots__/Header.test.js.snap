// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header mounts 1`] = `
<div
  className="sc-bZQynM sc-gzVnrw dJpxMi"
  height="125px"
/>
`;
